xls/dslx/run_routines.cc

```diff
@@ -279,6 +279,7 @@ absl::StatusOr<TestResult> ParseAndTest(std::string_view program,
                                     suffix)
               << "\n";
     failed += 1;
+    return status;
```

Comment:
I don't think we actually need to return anything here, this it's a `const` reference and we already take it as an argument (meaning that the caller is already able to check for its value)

Reply:
Yes, you're right, I'll remove it in next commit